code/game/objects/structures/xeno.dm

```diff
@@ -967,3 +999,59 @@ TUNNEL
 	user.fire_resist_modifier += 20
 	to_chat(user, "<span class='xenonotice'>We feel more vulnerable again.</span>")
 	qdel(src)
+
+
+///Proc called for repairable xeno structures
+/obj/proc/repair_xeno_structure(mob/living/carbon/xenomorph/M)
+
+	if(M.a_intent != INTENT_HELP) //We must be on help intent to repair
+		return FALSE
+
+	if(obj_integrity >= max_integrity) //If the structure is at max health (or higher somehow), no need to continue
+		return FALSE
+
+	if(!is_type_in_typecache(M, GLOB.xenorepaircastes)) //Check to see if we can actually repair
+		return FALSE
+
+	if(M.plasma_stored < 1) //You need to have at least 1 plasma to repair the structure
+		to_chat(M, "<span class='xenodanger'>We need plasma to repair [src]!</span>")
+		return FALSE
```

Comment:
All this structure junk should be 
A) structure subtype with the existing procs being used (better)
B) A flag
Because these procs just being standard deconstruction and whatever procs but on obj in a no no